### ISSUE TITLE
Dependencies between tasks

### ISSUE THREAD

**Issue by quazgar** [nextcloud/tasks#131]
it should be possible to have tasks depend on other tasks being completed, to move this app a bit more into a project management direction.

This could be implemented for example by restricting start or due dates on the dates of another task.

Such dependencies would then also imply a sorting order, even if no specific dates are given.

### Expected behaviour (details to be discussed)
Allow the user to specify a dependency as follows:

1. Tick a checkbox "Depends on other tasks"
2. From 3 dropdown menus, choose:
"This tasks cannot **{start | end}** before **other task** has been **{started | completed}**."
3. This should restrict both tasks' date options.
4. Make sure there are no loops in the dependency graph.

**Comment by witchof0x20** [comment]
I would be interested in at very least having a menu option that lets me specify that task A depends on tasks [B, C, D, ...]. This is probably the one feature from Taskwarrior I miss most after moving to Tasks.

**Comment by beedell-roke** [comment]
As I originally requested in [`issues/2185`](https://github.com/nextcloud/tasks/issues/2185#issue-1561309542), relationships between tasks exist almost everywhere else:

- #### GitLab

  ![A Screenshot](https://github.com/tasks/tasks/assets/42837531/da737ccb-b3fa-4da6-bb0f-089f86e9fde0) [^1]

  [^1]: [`gitlab.com/gitlab-org/gitlab/-/blob/c9925e46754a8de0327aa601e919eba851727aee/doc/user/project/issues/related_issues.md#add-a-linked-issue`](https://gitlab.com/gitlab-org/gitlab/-/blob/c9925e46754a8de0327aa601e919eba851727aee/doc/user/project/issues/related_issues.md#add-a-linked-issue:~:text=relates%20to,is%20blocked%20by)

- #### Notion

  ![A Screenshot](https://github.com/tasks/tasks/assets/42837531/2a1f4af2-2a31-4bb9-a603-8ccc0bf38f51) [^2]

  [^2]: [`notion.com/en-gb/help/tasks-and-dependencies`](https://www.notion.com/en-gb/help/tasks-and-dependencies)

However, Bugzilla provides the most comprehensive examples:

- Related

- Depends Upon

- Prevents

- Regresses

- Is Regressed By

**Comment by PeonyThePeon** [comment]
Related: [[Feature Request] Support RELTYPE=DEPENDS-ON #3188](https://github.com/nextcloud/tasks/issues/3188).
